src/jwplayer/players/shared/JwPlayerWrapperWithStrategyRules.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useContext } from 'react';
+import { JWPauseEvent, JWPlacementApi, JWPlayerApi, JWPlayEvent, JWPPlacementReadyResponse } from 'jwplayer/types';
+import { PlayerContext } from 'jwplayer/players/shared/PlayerContext';
+import { JwPlayerWrapperProps } from 'jwplayer/types';
+import { jwPlayerPlaybackTracker, triggerVideoMetric } from 'jwplayer/utils/videoTracking';
+import { recordVideoEvent, VIDEO_RECORD_EVENTS } from 'jwplayer/utils/videoTimingEvents';
+import JWEvents from 'jwplayer/players/shared/JWEvents';
+import addBaseTrackingEvents from 'jwplayer/players/shared/addBaseTrackingEvents';
+import useBeforeJwpWrapperRendered from 'jwplayer/utils/useBeforeJwpWrapperRendered';
+import useScript from 'jwplayer/utils/useScript';
+
+interface WindowJWPlayer extends Window {
+	jwplayer?: JWPlayerApi;
+	jwplacements?: JWPlacementApi;
+	jwDataStore?: Record<string, unknown>;
+	sponsoredVideos?: string[];
+}
+
+declare let window: WindowJWPlayer;
+
+const JwPlayerWrapperWithStrategyRules: React.FC<JwPlayerWrapperProps> = ({
+	config,
+	onReady,
+	getDismissed = () => false,
+	onComplete,
+	className,
+	shouldLoadSponsoredContentList = true,
+	jwPlayerContainerEmbedId = 'featured-video__player',
+	vastUrl,
+	parentClassName,
+}) => {
+	const strategyRulesPlacementId = '21rL5wJF';
+	const recommendationPlaylistId = 'FOhaD53w';
+
+	const registerEventHandlers = (playerInstance) => {
```

Comment:
can't have it shared between with and without strategy rules? 🤔

Reply:
I think I've been thinking about it and decided not to do it now. We're not sure if the strategy rules are the right solution and if they are, they'll replace the previous one, so maybe it's too early to share the code 🤔